packages/babel-generator/src/buffer.ts

```diff
@@ -27,23 +36,78 @@ function SourcePos(): SourcePos {
   };
 }
 
-const SPACES_RE = /^[ \t]+$/;
 export default class Buffer {
   constructor(map?: SourceMap | null) {
     this._map = map;
+
+    this._allocQueue();
   }
 
   _map: SourceMap = null;
   _buf = "";
+  _str = "";
+  _appendCount = 0;
   _last = 0;
-  _queue: Parameters<Buffer["_append"]>[] = [];
+  _queue: queueItem[] = [];
+  _queueCursor = 0;
 
   _position = {
     line: 1,
     column: 0,
   };
   _sourcePosition = SourcePos();
-  _disallowedPop: SourcePos | null = null;
+  _disallowedPop: SourcePos & { ok: boolean } = {
+    identifierName: undefined,
+    line: undefined,
+    column: undefined,
+    filename: undefined,
+    ok: false,
```

Comment:
I use this property instead to set and detect object as null. (This avoids creating and destroying temporary objects)
Yes, I also think the name is bad, but I'm not good at naming names😰, feel free to suggest!

Reply:
Maybe `objectReusable`? With a comment explaining that, to avoid deleting and re-creating objects, we reuse existing objects when they are not needed anymore.